tripper/triplestore.py

```diff
@@ -784,16 +506,14 @@ def infer_iri(obj):
     raise TypeError("cannot infer IRI from object {obj!r}")
 
 
-def function_id(func, length=4):
-    """Return a checksum for function `func`.
+def split_iri(iri: str) -> "Tuple[str, str]":
+    """Split iri into namespace and name parts and return them as a tuple."""
+    if "#" in iri:
+        namespace, name = iri.rsplit("#", 1)
+        return f"{namespace}#", name
 
-    The returned object is a string of hexadecimal digits.
+    if "/" in iri:
+        namespace, name = iri.rsplit("/", 1)
+        return f"{namespace}/", name
 
-    `length` is the number of bytes in the returned checksum.  Since
-    the current implementation is based on the shake_128 algorithm,
-    it make no sense to set `length` larger than 32 bytes.
-    """
-    # return hex(crc32(inspect.getsource(func).encode())).lstrip('0x')
-    return hashlib.shake_128(  # pylint: disable=too-many-function-args
-        inspect.getsource(func).encode()
-    ).hexdigest(length)
+    raise ValueError("all IRIs should contain a slash")
```

Comment:
```suggestion
    raise ValueError("all IRIs should contain a forward slash or a hash")
```

Reply:
Is that really true? "a#b" is not a valid IRI (even though that the current implementation will happily split it...).

Reply:
I was merely updating the error message according to the logic you were implementing in the function here. You were first checking for a hash, and then a forward slash, returning out of the function after each check, and so I put this logic into the error message for a bit of transparency for the developer/user who eventually reads it and needs to understand why something hasn't worked :)

Reply:
An IRI does not need a hash, although most ontologies uses a hash to separate the namespace from the name part.
A prominent exception is dcterms, which uses a slash...